### ISSUE TITLE
Log information should be timestamped

### ISSUE THREAD

**Issue by janisozaur** [OpenRCT2/OpenRCT2#4816]
**Commit/Build:** 6b4def9

Sample log line from current build:

`WARNING[/home/janisozaur/workspace/OpenRCT2/src/network/network.cpp:1502 (Client_Handle_OBJECTS)]: Object 1KWDTPI  has different checksum/flags (e648e7fb/9f55b03) than server (8698b1f4/9f55b03).`

For diagnosing some issues, it would be good to have a timestamp attached to each of the log lines, it could look like this (ISO 8601):

`2016-11-18T22:06:25 WARNING[/home/janisozaur/workspace/OpenRCT2/src/network/network.cpp:1502 (Client_Handle_OBJECTS)]: Object 1KWDTPI  has different checksum/flags (e648e7fb/9f55b03) than server (8698b1f4/9f55b03).`

We could optionally increase resolution of this information.

**Comment by marijnvdwerf** [comment]
Stupid question, but is it an idea to trim the path a bit?
`/home/janisozaur/workspace/OpenRCT2/src/` seems pretty useless to me.

**Comment by janisozaur** [comment]
@marijnvdwerf  `__FILE__`

**Comment by IntelOrca** [comment]
@janisozaur it already uses that, it would have to trim it at runtime or something.

**Comment by janisozaur** [comment]
How do you decide what do you trim it to?

**Comment by IntelOrca** [comment]
last usage of `src/`?

**Comment by RollingStar** [comment]
Relevant code: 

* [diagnostic.c](https://github.com/OpenRCT2/OpenRCT2/blob/06406de725da018ef5005de89a9d1f810d251706/src/diagnostic.c#L33)
* [diagnostic.h](https://github.com/OpenRCT2/OpenRCT2/blob/b29be9bfebe36f090b76b2553fc495a72c082e8a/src/diagnostic.h#L96)

So just add a C function to get the system time / date and add it to the first part of the log output?

**Comment by janisozaur** [comment]
Yes, there should already be some code to do that in `platform/`

**Comment by janisozaur** [comment]
Rejected in #23599
